### ISSUE TITLE
docker - ldap not ready

### ISSUE THREAD

**Issue by fvanderbiest** [georchestra/georchestra#1261]
When firing the composition,

```
cas_1               | -------------------------> LDAP Not Ready waiting... <-------------------------
ldap_1              | 56c47e48 connection_read(15): no connection!
cas_1               | -------------------------> LDAP Not Ready waiting... <-------------------------
cas_1               | -------------------------> LDAP Not Ready waiting... <-------------------------
```

Yet ldap container runs:

```
fvanderbiest@granier:~/workspace/georchestra/georchestra (master *$%)$ docker ps
CONTAINER ID        IMAGE                            COMMAND                  CREATED              STATUS              PORTS                              NAMES
3f1fe5a4d760        georchestra/security-proxy       "/docker-entrypoint.b"   About a minute ago   Up About a minute   0.0.0.0:8080->8080/tcp             georchestra_proxy_1
b60c6ebc2f2c        georchestra/ldapadmin            "/docker-entrypoint.b"   About a minute ago   Up About a minute   8080/tcp                           georchestra_ldapadmin_1
2b1c54306f49        camptocamp/courier-imap          "/root/start.sh"         About a minute ago   Up About a minute   143/tcp                            georchestra_courier-imap_1
01bfcfc40ee8        georchestra/geonetwork           "/docker-entrypoint.b"   About a minute ago   Up About a minute   0.0.0.0:1099->1099/tcp, 8080/tcp   georchestra_geonetwork_1
c46d3b40cab5        georchestra/cas                  "/docker-entrypoint.b"   About a minute ago   Up About a minute   8080/tcp                           georchestra_cas_1
dabbbbac92cc        georchestra/ldap                 "/bin/sh -c /start.sh"   About a minute ago   Up About a minute   389/tcp                            georchestra_ldap_1
6e12d521fa9d        georchestra/mapfishapp           "/docker-entrypoint.b"   About a minute ago   Up About a minute   8080/tcp                           georchestra_mapfishapp_1
85bfcc044a85        georchestra/header               "/docker-entrypoint.b"   About a minute ago   Up About a minute   8080/tcp                           georchestra_header_1
52ac49014e58        camptocamp/sqwebmail             "/root/start.sh"         About a minute ago   Up About a minute   0.0.0.0:8081->80/tcp               georchestra_webmail_1
e747e19179e8        georchestra/extractorapp         "/docker-entrypoint.b"   About a minute ago   Up About a minute   8080/tcp                           georchestra_extractorapp_1
6ba7f39ec92b        camptocamp/smtp-sink             "smtp-sink -c -d /hom"   About a minute ago   Up About a minute   25/tcp                             georchestra_smtp_1
2b3bf51cd04b        georchestra/geoserver-geofence   "/docker-entrypoint.b"   About a minute ago   Up About a minute   8080/tcp                           georchestra_geoserver_1
7b921106c325        georchestra/database             "/docker-entrypoint.s"   About a minute ago   Up About a minute   5432/tcp                           georchestra_database_1
49faf220236d        georchestra/ssh_data             "/bin/sh -c /root/sta"   About a minute ago   Up About a minute   0.0.0.0:2222->22/tcp               georchestra_geoserver_geodata_1
```

Docker version 1.10.0, build 590d5108
docker-compose version 1.6.0rc1, build d2556a1

**Comment by pmauduit** [comment]
That is an issue I already discussed with @vampouille, your LDAP should not contain the whole geOrchestra LDAP tree. The message printed by CAS makes you think that the LDAP container should not be running, but in fact when CAS tests the connection, it tries to look up users in the ou=users,dc=... and it probably does not find anything.

I encountered the issue last week, I had to remove my ldap container and its associated volume, then regenerate it.

**Comment by fvanderbiest** [comment]
Nope, this did not fix it.

**Comment by fvanderbiest** [comment]
The issue did not show up recently, closing.

**Comment by Gaetanbru** [comment]
Hi ! I have the same issue when i use docker and this [command lines](https://github.com/georchestra/docker): 
From 1 to 300 : 
cas_1  | ----------> LDAP Not Ready waiting... on ldap://ldap:389 with login : cn=admin,dc=georchestra,dc=org Searching entry with '(uid=*)' in ou=roles,dc=org (299/300)

![image](https://user-images.githubusercontent.com/16317988/28277630-d0555f52-6b1a-11e7-8f2d-e9e5e4f29575.png)
Now some java errors are display and system wait on Started......

![image](https://user-images.githubusercontent.com/16317988/28277669-ef60d58e-6b1a-11e7-9d04-f460aa5dc385.png)

Have you find this error again or a fix ?

**Comment by fvanderbiest** [comment]
Indeed the error sometimes shows up again after several docker-compose stop / start cycles.

I have not investigated it yet, but usually, removing the ldap container and its associated volumes is enough. I guess we have a permission problem with the LDAP image and its configuration files.

**Comment by Gaetanbru** [comment]
Thanks, i try this fix and come back later !

**Comment by Gaetanbru** [comment]
So, i remove : 
> docker rm -f docker_ldap_1
> docker volume rm docker_ldap_data
> docker volume rm docker_ldap_config

... and as write in #1656  i remove [this lines](https://github.com/georchestra/georchestra/blob/master/docker-compose.yml#L24-L34) from docker-compose.yml  

So i think i have erased too many things... when i restart service :

> docker-compose restart

I read new error : 
![image](https://user-images.githubusercontent.com/16317988/28323260-8828aaa4-6bd8-11e7-896b-d0f5e29536dd.png)

**Comment by Gaetanbru** [comment]
How can this work without ldap ?

**Comment by Vampouille** [comment]
> ... and as write in #1656 i remove this lines from docker-compose.yml

You don't need to remove lines in compose file. Just remove container with `docker-compose down` or `docker rm -f docker_ldap_1` and volumes with `docker volume rm docker_ldap_data docker_ldap_config`. Then you can restart composition with `docker-compose up`.

> Should i delete ldapadmin volumes and container to ?

no

> How can this work without ldap ?

You must have a valid ldap instance.

**Comment by Gaetanbru** [comment]
Thank you! I restore my docker-compose.yml and ` docker-compose up` works ! Wait result now....

**Comment by fvanderbiest** [comment]
LDAP container cold start logs:
```
run-parts: executing /docker-entrypoint.d/00-populate
Init with upstream entrypoint...Init: OK
Starting slapd daemon in background...Started: OK
Waiting for LDAP deamon to start...Waiting: OK
Populating LDAP tree...
SASL/EXTERNAL authentication started
SASL username: gidNumber=0+uidNumber=0,cn=peercred,cn=external,cn=auth
SASL SSF: 0
adding new entry "cn=module,cn=config"

adding new entry "olcOverlay=memberof,olcDatabase={1}hdb,cn=config"

adding new entry "ou=users,dc=georchestra,dc=org"

adding new entry "uid=testuser,ou=users,dc=georchestra,dc=org"

adding new entry "uid=testreviewer,ou=users,dc=georchestra,dc=org"

adding new entry "uid=testeditor,ou=users,dc=georchestra,dc=org"

adding new entry "uid=testadmin,ou=users,dc=georchestra,dc=org"

...

adding new entry "cn=USER,ou=roles,dc=georchestra,dc=org"

adding new entry "ou=orgs,dc=georchestra,dc=org"

adding new entry "o=psc,ou=orgs,dc=georchestra,dc=org"

adding new entry "cn=psc,ou=orgs,dc=georchestra,dc=org"

Populating LDAP tree: OK
Creating indexes...
SASL/EXTERNAL authentication started
SASL username: gidNumber=0+uidNumber=0,cn=peercred,cn=external,cn=auth
SASL SSF: 0
modifying entry "olcDatabase={1}hdb,cn=config"

modifying entry "olcDatabase={1}hdb,cn=config"

Indexes: OK
Waiting for LDAP to stop...Waiting: OK
run-parts: executing /docker-entrypoint.d/01-indexes

WARNING!
Runnig as root!
There's a fair chance slapd will fail to start.
Check file permissions!

LDAP indexed
598440af @(#) $OpenLDAP: slapd  (Jan 16 2016 23:00:08) $
	root@chimera:/tmp/buildd/openldap-2.4.40+dfsg/debian/build/servers/slapd
598440af slapd starting
```
We should not run slapindex (from `01-indexes`) as root.

**Comment by fvanderbiest** [comment]
This issue should be fixed on 16.12 and later with #1765.
